### PR TITLE
feature(housekeeping stats): add housekeeping stats test to artifact tests

### PR DESCRIPTION
**Pull request by juliayakovlev**
Add housekeeping stats test to artifact tests

Tested on configurations:
- ami
- amazon2
- ubuntu2004
- docker

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: artifacts_test.py [hunk 1]
```diff
@@ -16,18 +16,68 @@
...
+    #
+    #     return row[0] if row else 0
+
     @property
     def node(self):
         if self.db_cluster is None or not self.db_cluster.nodes:
             raise ValueError('DB cluster has not been initiated')
         return self.db_cluster.nodes[0]
 
+    @property
```
Comment (bentsi) [inline artifacts_test.py:76]: use `@cached_property`

File: artifacts_test.py [hunk 2]
```diff
@@ -16,18 +16,68 @@
...
+    #
+    #     super().tearDown()
+
+    # TODO: uncomment check_version when problem with SSL connection to MYSQL will be solved
+    # def check_version(self, prev_id: int, new_row_expected: bool) -> int:
+    #     """
+    #     Validate reported version
+    #     prev_id: check if new version is created
+    #     """
+    #     row = self.housekeeping.get_most_recent_record(f"SELECT id, version, ip FROM {self.CHECK_VERSION_TABLE} "
```
Comment (bentsi) [inline artifacts_test.py:47]: Good style is:
If function/method have more than one argument, call the function with key-word arguments (for example foo(x=10, y=20) is correct call  and foo(10, 20) is not)
so `get_most_recent_record` should be called as:
`get_most_recent_record(query=f"...."`

File: artifacts_test.py [hunk 3]
```diff
@@ -16,18 +16,68 @@
...
+    #     self.housekeeping = HousekeepingDB.from_keystore_creds()
+    #     self.housekeeping.connect()
+    #
+    # def tearDown(self) -> None:
+    #     self.housekeeping.close()
+    #
+    #     super().tearDown()
+
+    # TODO: uncomment check_version when problem with SSL connection to MYSQL will be solved
+    # def check_version(self, prev_id: int, new_row_expected: bool) -> int:
```
Comment (bentsi) [inline artifacts_test.py:42]: name is misleading. Which version? Where it checks? What is the check? find better name

File: artifacts_test.py (hunk 2 above)
Comment (bentsi) [inline artifacts_test.py:47]: also please check that `self.node.uuid` is not None, if there is no UUID  smth bad happened, so need to assert in that case

**Review by bentsi — CHANGES_REQUESTED** [review]
please address comments

File: artifacts_test.py (hunk 1 above)
Comment (juliayakovlev) [reply artifacts_test.py:76]: changed

File: artifacts_test.py (hunk 2 above)
Comment (juliayakovlev) [reply artifacts_test.py:47]: changed

File: artifacts_test.py (hunk 3 above)
Comment (juliayakovlev) [reply artifacts_test.py:42]: changed to `check_scylla_version_in_housekeepingdb`

File: artifacts_test.py (hunk 2 above)
Comment (juliayakovlev) [reply artifacts_test.py:47]: added assertion

**Comment by juliayakovlev** [comment]
@bentsi  please, review

**Comment by roydahan** [comment]
ping

**Review by dkropachev — APPROVED** [review]

File: sdcm/utils/net.py [hunk 4]
```diff
@@ -15,3 +22,31 @@ def get_my_ip():
...
+                raise
+        except (requests.exceptions.RequestException, ValueError) as error:
+            LOGGER.error("Failed to get public IP address from '%s'. Error: %s", hostname, str(error))
+            if idx == len(hostnames) - 1:
+                raise
+
+    # Added empty return due pylint validation error:
+    # R1710: Either all return statements in a function should return an expression, or none of them should.
+    # (inconsistent-return-statements)
+    return ''
```
Comment (bentsi) [inline sdcm/utils/net.py:52]:
```suggestion
def get_my_public_ip() -> str:
    hostnames = ['https://api4.my-ip.io/ip', "https://api.ipify.org", 'https://ip4.seeip.org/ip',
                 'http://ipv4.icanhazip.com']

    for hostname in hostnames:
        result = requests.get(hostname, timeout=10)
        if result.ok:
            try:
                ip_address = result.text.strip()                
                ipaddress.ip_address(ip_address)  # validating that we got IP address
                return ip_address
            except ValueError:
                continue
    return ""
```

**Review by bentsi — APPROVED** [review]

**Pull request merged**